engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -769,7 +769,11 @@ public void processHeartbeatResponse(ContainerHeartbeatResponse rsp)
     }
 
     if (rsp.committedWindowId != lastCommittedWindowId) {
+
       lastCommittedWindowId = rsp.committedWindowId;
+
+      bufferServer.purge(lastCommittedWindowId);
```

Comment:
Please address the question regarding the windowId. If it isn't decremented, aren't we blowing away the checkpoint window?